src/methods/has.js

```diff
@@ -61,7 +61,6 @@ export function HasName(realm: Realm, ast: BabelNodeExpression): boolean {
 // ECMA262 7.3.10
 export function HasProperty(realm: Realm, O: ObjectValue | AbstractObjectValue, P: PropertyKeyValue): boolean {
   // 1. Assert: Type(O) is Object.
-  invariant(O instanceof ObjectValue || O instanceof AbstractObjectValue, "expected object");
 
```

Comment:
Did you delete this because of the static signature? I think there's value in the assertion just because Flow isn't sound. Maybe `invariant(!(O.mightNotBeObject));`